admin/win/msi/make-msi.bat.in

```diff
@@ -53,6 +53,12 @@ if "%WISUBSTG%" == "" (
   echo :: Did not find WiLangId.vbs ^(environment variable 'WILANGID'^), the resulting MSI will be in English by default
 ))
 
+REM generate a ProductCode to use in the resulting MSI to ensure that the embedded
+REM transforms don't break managed deployments (see #8610)
+REM the uuidgen tool is provided by the Visual Studio CLI tools
+for /f %%i in ('uuidgen -c') do set PRODUCTCODE=%%i
```

Comment:
why are we generating a new `PRODUCTCODE for each new build ?

Reply:
We always did (that's what `<Product Id="*"` does).  However that product code apparently is set during the linking stage and not during compilation, which resulted in each language-specific build ending up with a different own product code.  As these are embedded as transforms that get loaded during startup, this breaks some managed installation scripts...

Reply:
in addition, I was pointed to https://learn.microsoft.com/en-us/windows/win32/msi/changing-the-product-code
explains when the `PRODUCTCODE` has to be changed